### DISCUSSION TITLE
How to update my docker setup?

### DISCUSSION THREAD

**Discussion by nooobcoder** [outline/outline#2209] [Self Hosting]
I have cloned the repo made my own modifications as per need and docs. Created my .env file also modified docker-compose.yml file. The big question is how do i update the app when an update is pushed, without losing any data.

As far i know, the data is being stored in the following folders,
`./fakes3` and `.:/opt/outline`.

Please guide me on updating the setup, also make docker friendly images, please. **Also, remove the constraint of slack, google oauth based logins for self hosted setups. If i am doing self hosted setups then I have a private ip ofcourse, and private ips cause headache in oauth especially for google oauth. Thank You**

I am pasting my docker-compose settings for reference.

![image](https://user-images.githubusercontent.com/50350828/121558211-18042600-ca33-11eb-90dc-9fc76ff8d83e.png)

`version: "3"
services:
  redis:
    image: redis
    ports:
      - "6479:6379"
  postgres:
    image: postgres
    ports:
      - "5532:5432"
    environment:
      POSTGRES_USER: user
      POSTGRES_PASSWORD: pass
      POSTGRES_DB: outline
  s3:
    image: lphoward/fake-s3
    ports:
      - "4569:4569"
    volumes:
      - ./fakes3:/fakes3_root
  outline:
    image: outline:v001
    command: yarn dev
    build:
      context: .
      dockerfile: Dockerfile
      args:
        pull: 1
    ports:
      - "7777:3000"
    volumes:
      - .:/opt/outline
    depends_on:
      - postgres
      - redis
      - s3`

**Comment by tommoor** [comment]
If you haven't customized Outline you should change your docker-compose file to use the official docker build available here:

https://hub.docker.com/repository/docker/outlinewiki/outline
